crates/rpc-types/src/error.rs

```diff
@@ -0,0 +1,98 @@
+//! RPC errors specific to OP.
+
+use alloy_rpc_types_eth::error::EthRpcErrorCode;
+use revm_primitives::{InvalidTransaction, OptimismInvalidTransaction};
+use std::fmt::Display;
+
+/// Optimism specific errors.
+#[derive(Debug, thiserror::Error)]
+pub enum OpEthApiError<EthApiError: Display> {
+    /// L1 ethereum error.
+    #[error(transparent)]
+    Eth(#[from] EthApiError),
+    /// Thrown when calculating L1 gas fee.
+    #[error("failed to calculate l1 gas fee")]
+    L1BlockFeeError,
+    /// Thrown when calculating L1 gas used
+    #[error("failed to calculate l1 gas used")]
+    L1BlockGasError,
+    /// Wrapper for [`revm_primitives::InvalidTransaction`].
+    #[error(transparent)]
+    InvalidTransaction(OptimismInvalidTransactionError),
```

Comment:
would it be ok to just incorporate these variants in a new error type and leave the rest in reth?